src/rosdiscover/config.py

```diff
@@ -57,14 +58,25 @@ def from_dict(cls, dict_: Mapping[str, Any]) -> 'Config':
             raise ValueError("expected 'sources' to be a list")
         if not isinstance(dict_['launches'], list):
             raise ValueError("expected 'launches' to be a list")
-
+        if isinstance(dict_['launches'][0], dict):
+            launch_args_provided = True
+        elif isinstance(dict_['launches'][0], str):
+            launch_args_provided = False
+        else:
+            raise ValueError("expected elements of 'launches' to be str or dict")
+            
         has_environment = 'environment' in dict_
         if has_environment and not isinstance(dict_['environment'], dict):
             raise ValueError("expected 'environment' to be a mapping")
 
         image: str = dict_['image']
         sources: Sequence[str] = dict_['sources']
-        launches: Sequence[str] = dict_['launches']
+        launches_inputs: Sequence[str] = dict_['launches']
```

Comment:
This type should probably be `Sequence[Any]` now (or a more specific albeit verbose union type).

Reply:
fixed